kyo-core/shared/src/main/scala/kyo/Async.scala

```diff
@@ -440,16 +444,18 @@ object Async:
       *   The duration to sleep
       */
     def sleep(d: Duration)(using Frame): Unit < Async =
-        IO {
-            val p = IOPromise[Nothing, Unit]()
-            if d.isFinite then
-                Timer.schedule(d)(p.completeUnit(Result.success(()))).map { t =>
-                    IO.ensure(t.cancel.unit)(get(p))
-                }
-            else
-                get(p)
-            end if
-        }
+        if d == Duration.Zero then ()
```

Comment:
I noticed that `Retry` ends up calling `Async.sleep` with a zero duration for the first retry with the default policy.